controllers/clusteroperator_controller.go

```diff
@@ -0,0 +1,160 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"fmt"
+
+	configv1 "github.com/openshift/api/config/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/klog/v2"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	"github.com/openshift/library-go/pkg/config/clusteroperator/v1helpers"
+)
+
+const (
+	clusterOperatorName = "cloud-controller-manager"
```

Comment:
Does this follow convention? This is the name of the operand rather than the operator?

Reply:
You are right. Just looking on the list there is not much convention, but the name I picked is definitely for the operand. Changing it to `openshift-cloud-controller-manager` will probably be best.
```
NAME                                       VERSION   AVAILABLE   PROGRESSING   DEGRADED   SINCE
authentication                             4.6.9     True        False         False      146m
cloud-controller-manager                   0.0.1     True        False         False      127m
cloud-credential                           4.6.9     True        False         False      177m
cluster-autoscaler                         4.6.9     True        False         False      171m
config-operator                            4.6.9     True        False         False      172m
console                                    4.6.9     True        False         False      157m
csi-snapshot-controller                    4.6.9     True        False         False      172m
dns                                        4.6.9     True        False         False      170m
etcd                                       4.6.9     True        False         False      170m
image-registry                             4.6.9     True        False         False      162m
ingress                                    4.6.9     True        False         False      162m
insights                                   4.6.9     True        False         False      172m
kube-apiserver                             4.6.9     True        False         False      168m
kube-controller-manager                    4.6.9     True        False         False      169m
kube-scheduler                             4.6.9     True        False         False      169m
kube-storage-version-migrator              4.6.9     True        False         False      162m
machine-api                                4.6.9     True        False         False      165m
machine-approver                           4.6.9     True        False         False      171m
machine-config                             4.6.9     True        False         False      170m
marketplace                                4.6.9     True        False         False      170m
monitoring                                 4.6.9     True        False         False      161m
network                                    4.6.9     True        False         False      172m
node-tuning                                4.6.9     True        False         False      172m
openshift-apiserver                        4.6.9     True        False         False      165m
openshift-controller-manager               4.6.9     True        False         False      169m
openshift-samples                          4.6.9     True        False         False      164m
operator-lifecycle-manager                 4.6.9     True        False         False      171m
operator-lifecycle-manager-catalog         4.6.9     True        False         False      171m
operator-lifecycle-manager-packageserver   4.6.9     True        False         False      165m
service-ca                                 4.6.9     True        False         False      172m
storage                                    4.6.9     True        False         False      171m
```

Reply:
Looks like `openshift-cloud-controller-manager` or just `cloud-controller-manager` would be suitable, I think looking at the list I'd lean towards the latter. You updated it to include operator in the name which doesn't seem to fit with the list above

Reply:
My bad, should be ok now :D

Reply:
I prefer `cloud-controller-manager` too

Reply:
Let's go with `cloud-controller-manager`, sorry for the confusion @Danil-Grigorev

Reply:
NP, it is sealed.